### PR TITLE
Feature 7.1 - audio timestamps

### PR DESCRIPTION
**Pull request by LiamOdero**
### TLDR
-  Reworked visual aspects of the player and the AudioTool
-  Adds the ability to both add and delete timestamps
-  Allowed `MediaAudio` to persist with the page

**Notes:**
- For now, timestamps are affixed to the associated player since this PR focuses on merely implementing functionality. This can (and likely will be) changed once features for manually adjusting where `GUIMedia` on the page lies

### Visual Changes:
**New Player Look Example:**
image
**New Audio Tool Look Example:**
image
**Audio Tool when a Player has been Clicked Example:**
image

### Changes:
- Changed implementation of `ToolBarController.insertAudio`, `AudioModifier.addMedia` to allow persistence
- Fully implemented `ToolBarController.modifyTimestamp`, `ToolBarController.createHyperlink`, `AudioModifier.modifyMedia` to allow creation of timestamp hyperlinks

- Created a very basic implementation of `GUIHyperlink` to support adding hyperlinks to the page

- Fully implemented `GUIAudio.createTimestamps` and `GUIAudio.mediaUpdated` to visually support adding timestamps
- Creating a variety of getter/setter functions in `GUIAudio` for use by different methods
- Modified `GUIAudio.createInterface` for new look
- Implemented `GUIAudio.echoClick` to account for UI controls seemingly consuming mouse events

- Implemented `AudioTool.setSelectedAudio` to handle when a player is clicked for the `AudioTool`
- Modified `AudioTool.getSettingsGUI` for new look

- Restructured exception handling to account for changes to how `GUIMedia` is added to the page

### CODE REVIEW

**Review by Uruk-Haiku — APPROVED** [review]
Great work! I love how thorough your test coverage is. In some of the longer files, I would recommend adding headers and spacing to make it clear what does what, but overall your comments look great! As well, your style adherence is great! The only major complaint I have is that I think some Javadoc for the larger methods would be a good idea, but it might be redundant anyway. Great stuff!

**Review by Uruk-Haiku — APPROVED** [review]
Looks fresh!

**Review by JeremyXie123 — APPROVED** [review]
Great work. I must agree with Ben, it may be a good idea to add Javadocs, particularly to the functions who's operation is not immediately intuitive. This way you can avoid making docs for each and every function (as there are quite a few).

**Comment by LiamOdero** [comment]
> Great work! I love how thorough your test coverage is. In some of the longer files, I would recommend adding headers and spacing to make it clear what does what, but overall your comments look great! As well, your style adherence is great! The only major complaint I have is that I think some Javadoc for the larger methods would be a good idea, but it might be redundant anyway. Great stuff!

> Great work. I must agree with Ben, it may be a good idea to add Javadocs, particularly to the functions who's operation is not immediately intuitive. This way you can avoid making docs for each and every function (as there are quite a few).

Added improved documentation to address feedback
